src/fragments/lib/auth/js/advanced.mdx

```diff
@@ -370,6 +370,39 @@ Note: To work with Service Interface Objects, your Amazon Cognito users' [IAM ro
 
 </Callout>
 
+## Custom Token providers
+
+You can use your own token provider across the library, for example, to use it with AppSync when OIDC is the Auth mode. You must supply the token provider to Amplify via the `Amplify.configure` method call. Below, you can see sample code of how such a custom provider can be built to achieve the use case.
```

Comment:
Should we callout that this it for non-cognito use cases? What about something like this?

```suggestion
Create a custom Auth token provider for situations where you would like provide your own tokens for a service. For example, using OIDC Auth with AppSync.  You must supply the token provider to Amplify via the `Amplify.configure` method call. Below, you can see sample code of how such a custom provider can be built to achieve the use case.
```